packages/react-router/src/router.ts

```diff
@@ -1890,235 +1926,251 @@ export class Router<
+
+                  const staleAge = preload
+                    ? route.options.preloadStaleTime ??
+                      this.options.defaultPreloadStaleTime ??
+                      30_000 // 30 seconds for preloads by default
+                    : route.options.staleTime ??
+                      this.options.defaultStaleTime ??
+                      0
+
+                  const shouldReloadOption = route.options.shouldReload
+
+                  // Default to reloading the route all the time
+                  // Allow shouldReload to get the last say,
+                  // if provided.
+                  const shouldReload =
+                    typeof shouldReloadOption === 'function'
+                      ? shouldReloadOption(getLoaderContext())
+                      : shouldReloadOption
+
+                  updateMatch(matchId, (prev) => ({
+                    ...prev,
+                    preload:
+                      !!preload &&
+                      !this.state.matches.find((d) => d.id === matchId),
+                  }))
+
+                  const fetchWithRedirectAndNotFound = async () => {
+                    try {
+                      await fetchAndResolveInLoaderLifetime()
+                    } catch (err) {
+                      checkLatest()
+                      handleRedirectAndNotFound(getMatch(), err)
+                    }
+                  }
 
-                // This is where all of the stale-while-revalidate magic happens
-                const age = Date.now() - match.updatedAt
-
-                const staleAge = preload
-                  ? route.options.preloadStaleTime ??
-                    this.options.defaultPreloadStaleTime ??
-                    30_000 // 30 seconds for preloads by default
-                  : route.options.staleTime ??
-                    this.options.defaultStaleTime ??
-                    0
-
-                const shouldReloadOption = route.options.shouldReload
-
-                // Default to reloading the route all the time
-                // Allow shouldReload to get the last say,
-                // if provided.
-                const shouldReload =
-                  typeof shouldReloadOption === 'function'
-                    ? shouldReloadOption(loaderContext)
-                    : shouldReloadOption
-
-                matches[index] = match = {
-                  ...match,
-                  preload:
-                    !!preload &&
-                    !this.state.matches.find((d) => d.id === match.id),
-                }
+                  // If the route is successful and still fresh, just resolve
+                  const { status, invalid } = getMatch()
+
+                  if (
+                    status === 'success' &&
+                    (invalid || (shouldReload ?? age > staleAge))
+                  ) {
+                    ;(async () => {
+                      try {
+                        await fetchWithRedirectAndNotFound()
```

Comment:
Does this still need to be within a try-catch block considering `fetchWithRedirectAndNotFound` already uses one within the function definition?
Asking since we don't use a try-catch block below at `if (status !== 'success') {`.

Reply:
... My gut says yes, because there can be synchronous errors that we need to catch, not just async ones